impl/src/context.rs

```diff
@@ -136,6 +137,7 @@ pub struct CrateContext {
   pub source_details: SourceDetails,
   pub sha256: Option<String>,
   pub registry_url: String,
+  pub is_proc_macro: bool,
```

Comment:
Is this the right place for this? IIUC, "proc_macro"-ness is a property of a target within the crate, not  the crate itself. I expect that a proc macro should contain a target with the corresponding kind.

Reply:
This is meant to go along with `pub lib_target_name: Option<String>,` which is currently set by both `lib` and `proc-macro` targets but the distinction is important to be able to identify later. Though the placement of that variable doesn't quite convey that so I can update that if you think this is something that could stay.

Reply:
I've updated the placement of this variable and made a comment. Hopefully this makes it more clear.